### PR TITLE
Add build instructions for Fedora.

### PR DESCRIPTION
**Pull request by rl2939**

### CODE REVIEW

File: docs/BUILDING.md [hunk 1]
```diff
@@ -101,6 +101,26 @@ sudo apt install cmake libdbus-1-dev libgl1-mesa-dev liblo-dev libfftw3-dev libm
...
+# system libraries
+sudo dnf install libarchive libsamplerate speexdsp
+```
+
+Dependencies for vendored libraries:
+
+```
+# common
+sudo dnf install cmake dbus file fftw mesa-libGL liblo libsndfile libX11 libXcursor libXext libXrandr python3
+# nedeed by vendored libraries
```
Comment (dromer) [inline docs/BUILDING.md:120]: typo

Comment (rl2939) [reply docs/BUILDING.md]: Fixed, check new commits.

**Comment by rl2939** [comment]
Is there anything else that needs to be done before this can be merged?

**Comment by dromer** [comment]
I haven't run Fedora in nearly a decade. Can you confirm 100% that these are all the required dependencies to do a full build? :)

**Comment by rl2939** [comment]
I am able to build using vendored libraries. I am unable to test with system libraries (`SYSDEPS=true`), as the makefile says that "jansson dependency not installed/available.  Stop.", despite jansson already being installed.

**Comment by dromer** [comment]
You say the jansson dependency is already installed, but shouldn't the instructions explicitly install it?
That should be the development headers aka `jansson-devel` (from what I can tell).

**Comment by rl2939** [comment]
Adding `jansson-devel` does not change anything. 
```
$ make SYSDEPS=true
Makefile:53: *** jansson dependency not installed/available.  Stop.
```
Because I can't get it to work, should I remove the "Dependencies for using system libraries" section? There will only be a section for vendored libraries for Fedora.

**Comment by dromer** [comment]
@rl2939 hmm odd. but yeah if these instructions don't work then maybe better to omit them and only use the ones for the vendored libs.

**Comment by rl2939** [comment]
Done.

**Pull request merged**